crates/pep440-rs/src/version_specifier.rs

```diff
@@ -340,9 +340,18 @@ impl Serialize for VersionSpecifier {
 }
 
 impl VersionSpecifier {
+    /// Create a new version specifier from an operator and a version.
+    ///
+    /// Warning: This function is not recommended for general use. It is intended for use in
+    /// situations where the operator and version are known to be valid. For general use, prefer
+    /// [`VersionSpecifier::from_pattern`].
+    pub fn new(operator: Operator, version: Version) -> Self {
```

Comment:
I see, reading the code below, this does seem to actually be used to build a `VersionSpecifier` that is otherwise considered invalid.

I find this to be somewhat odd and an unfortunate breakage in encapsulation. But I think I see why you're doing it. As a mitigation, what do you think about changing `from_pattern` back to `new`, and renaming your `new` routine something like, `from_possibly_invalid_parts` or something similarish in name? I would also add to the `VersionSpecifier` docs that callers cannot necessarily rely on the operator/version combination to be valid. That is, it is an intended and supported use case that a `VersionSpecifier` is a _superset_ of what's in PEP 440.

Also, are we sure that things that were previously not allowed but now are (like `VersionSpecifier::new(Operator::LessThanEqual, version_with_local)`) have expected behavior in the various routines on `VersionSpecifier`? Previously, the implementation could assume that the invalid combinations never exist, but that assumption no longer holds. (It's possible everything is fine as-is, but maybe some increased test coverage would be helpful here.)

Reply:
I'll do all these things.

Reply:
It's also possible that I can remove the actual invalid ones.

Reply:
Okay, I removed all the invalid versions, re-added the validation, and threaded through the errors.